Posts/2022/07/cheat-sheet-console-experience.md

```diff
@@ -0,0 +1,227 @@
+
+## Prompt
+
+Want to customize your prompt to be more colorful / fancy / whatever else you want it to do?
+
+Well, all you need to do is override the function named "prompt" and put it in your profile and that's that.
+Don't know how or want to borrow from others to make your life easier?
+
+Give [Powerline (for PowerShell)](https://github.com/Jaykul/PowerLine) a chance.
+They have some [fancy examples](https://github.com/Jaykul/PowerLine/tree/master/Source/Examples) as well!
+
+![A colorful command prompt](./prompt.png)
+
+## Dynamic Tab Completion
+
+With the previous notes on Tab Completion, you already saw how to get better tab completion.
+But PowerShell is still not reading your mind when it comes to the values provided - if a command doesn't offer it, you're out of luck.
+Right?
+
+Well no, there's tools to fix that.
+There are some options, but the simple-most is probably from the [PSFramework project](https://psframework.org/documentation/documents/psframework/tab-completion.html).
+To install it, run this line (once):
+
+```powershell
+Install-Module PSFramework
+```
+
+Then you can add the magic to your profile.
+Here is a quick example on how to add values to the "-Tenant" parameter on "Set-AzContext":
+
+```powershell
+Register-PSFTeppScriptblock -Name AZ.Tenant -ScriptBlock {
+  (Get-AZTenant).DefaultDomain
+}
+Register-PSFTeppArgumentCompleter -Command Set-AZContext -Parameter Tenant -Name AZ.Tenant
+```
+
+## The Clipboard
+
+The clipboard is always a handy tool to interactively cross over between applications.
+Now if only there were commands in PowerShell to do so ...
+
+```powershell
+Get-Clipboard
+Set-Clipboard
+```
+
+And since we're all about being lazy, there's aliases for that: "gcb" and "scb".
+On that note, if you want to paste multiple columns into Excel, you want to use the tab delimiter.
+Don't do manual labor though, here's the easy way to get data ready to paste to Excel:
+
+```powershell
+dir | ConvertTo-Csv -Delimiter "`t" | scb
+```
+
+That "ConvertTo-Csv" is way too much text though. Wouldn't it be nice to make that shorter?
+
+## Aliases
+
+In PowerShell, there is an easy way to be lazy: Aliases.
+Use their power to abbreviate your commonly used commands.
+Then put it in your $profile so you don't have to remember to add them.
+
+```powershell
+Set-Alias ctc ConvertTo-Csv
+```
+
+And now that previous line can be shortened to:
+
+```powershell
+dir | ctc -d "`t" | scb
+```
+
+Where is that "-d" coming from though?
+
+## Short Parameter Names
+
+You know all these commandline tools that have a long and a short notation for their parameters?
+Like where you can either specify "--help" or "-h"?
+
+Well, PowerShell takes that a step further:
+You only need to type a parameter so far until there's only one possibility what parameter it could be.
+
+Using the example above with "ConvertTo-Csv", there is only a single parameter that starts with "D", so that's enough to specify it.
```

Comment:
```suggestion
Using the example above with `ConvertTo-Csv`, there is only a single parameter that starts with "D", so that's enough to specify it.
```